src/TAProcessor.cpp

```diff
@@ -87,11 +87,11 @@ TAProcessor::conf(const appdal::ReadoutModule* conf)
 
   for (auto algo : tc_algorithms)  {
     TLOG() << "Selected TC algorithm: " << algo->UID();
-    std::unique_ptr<triggeralgs::TriggerCandidateMaker> maker = make_tc_maker(algo->class_name());
+    std::shared_ptr<triggeralgs::TriggerCandidateMaker> maker = make_tc_maker(algo->class_name());
```

Comment:
make_tc_maker still returns a unique_ptr, no?

Reply:
yes, but that is a typical pattern for a factory. It returns a unique pointer which is the more strict type of signature, but it is allowed to transform it right away to a shared ptr, if the user code requires it.